Zend/zend_fibers.c

```diff
@@ -537,16 +537,12 @@ ZEND_METHOD(Fiber, suspend)
 
 	if (UNEXPECTED(!fiber)) {
 		zend_throw_error(zend_ce_fiber_error, "Cannot suspend outside of a fiber");
-		return;
+		RETURN_THROWS();
 	}
 
-	if (UNEXPECTED(fiber->status != ZEND_FIBER_STATUS_RUNNING)) {
-		if (fiber->status == ZEND_FIBER_STATUS_SHUTDOWN) {
-			zend_throw_error(zend_ce_fiber_error, "Cannot suspend in a force closed fiber");
-		} else {
-			zend_throw_error(zend_ce_fiber_error, "Cannot suspend in a fiber that is not running");
-		}
-		return;
+	if (UNEXPECTED(fiber->status == ZEND_FIBER_STATUS_SHUTDOWN)) {
+		zend_throw_error(zend_ce_fiber_error, "Cannot suspend in a force closed fiber");
+		RETURN_THROWS();
```

Comment:
You might want to retain this as an assertion, i.e. `ZEND_ASSERT(fiber->status == ZEND_FIBER_STATUS_RUNNING);` below.